Code/Entities/ClimbableOneWayPlatform.cs

```diff
@@ -230,35 +274,99 @@ public class ClimbableOneWayPlatform : Entity {
 
                 // inject ourselves to jump over the "Speed.Y < 0f" check, and put this back
                 cursor.EmitDelegate<Func<Player, bool>>(self => {
-                    ClimbableOneWayPlatform platform = collideFirstOutside(self, self.Position + new Vector2((int) self.Facing * 2, 0), self.Facing == Facings.Left);
+                    ClimbableOneWayPlatform platform = collideFirstOutside(self, self.Position + new Vector2((int)self.Facing * 2, 0), self.Facing == Facings.Left);
                     return platform != null && platform.climbJumpGrabCooldown <= 0f;
                 });
                 cursor.Emit(OpCodes.Brtrue, afterCheck);
                 cursor.Emit(OpCodes.Ldarg_0);
             }
         }
 
-        private static void modPlayerClimbJump(On.Celeste.Player.orig_ClimbJump orig, Player self) {
+        //grabs the speed and stamina if applicable at the earliest poossible time
+        private static void Customization_modPlayerClimbBegin(On.Celeste.Player.orig_ClimbBegin orig, Player self)
+        {
+            ClimbableOneWayPlatform platform = collideFirstOutside(self, self.Position + new Vector2((int)self.Facing * 2, 0), self.Facing == Facings.Left);
+            if (platform != null)
+            {
+                //store initial speed and stamina if needed   
+                if (platform.hasMomentumCarrying)
+                {
+                    platform.initialSpeed = self.Speed.X;
+                    platform.startTimer();
+                }
+                if (platform.initialStamina >= 0f)

```

Comment:
I would check for the stamina behavior instead of this - this seems to rely on the fact that initialStamina is initialized to -1 if the behavior is not "conserve", which is a bit weird ^^'

Reply:
Checked the enum instead